crates/kornia-tensor/src/tensor.rs

```diff
@@ -618,6 +618,84 @@ impl<T, const N: usize, A: TensorAllocator> Tensor<T, N, A> {
         }
     }
 
+    /// Checks if the Tensor has a standard contiguous layout according to its `shape` and `strides`.
+    ///
+    /// # Returns
+    ///
+    /// boolean, true if contiguous and false if not
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use kornia_tensor::{Tensor, CpuAllocator};
+    /// let data: Vec<u8> = vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+    /// let mut t = Tensor::<u8, 3, CpuAllocator>::from_shape_vec([2, 2, 3], data, CpuAllocator).unwrap();
+    /// // arbitrary incorrect stride
+    /// t.strides = [10, 5, 1];
+    /// assert!(!t.is_standard_layout());
+    /// ```
+    pub fn is_standard_layout(&self) -> bool {
+        let mut expected_stride: usize = 1;
+        for (&dim, &stride) in self.shape.iter().rev().zip(self.strides.iter().rev()) {
+            if stride != expected_stride {
+                return false;
+            }
+            expected_stride = expected_stride.saturating_mul(dim);
+        }
+        true
+    }
+
+    /// Copy Tensor storage data into contiguous memory if not already
+    ///
+    /// # Returns
+    ///
+    /// A new Tensor with contiguous storage
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use kornia_tensor::{Tensor, CpuAllocator};
+    /// let data: Vec<u8> = vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+    /// let mut t = Tensor::<u8, 3, CpuAllocator>::from_shape_vec([2, 2, 3], data.clone(), CpuAllocator).unwrap();
+    /// // altering strides
+    /// t.strides = [1, 6, 2];
+    /// assert!(!t.is_standard_layout());
+    /// let t2 = t.to_standard_layout(CpuAllocator);
+    /// assert!(t2.is_standard_layout());
+    /// ```
+    pub fn to_standard_layout(&self, alloc: A) -> Self
+    where
+        T: Clone,
+    {
```

Comment:
should we check wether it's altready in standard layout ? or at least especify in the docs